src/app_service/service/community/service.nim

```diff
@@ -263,6 +264,39 @@ QtObject:
       error "Error creating community", msg = e.msg
       raise newException(RpcException, "Error creating community: " & e.msg)
 
+  proc editCommunity*(
+      self: Service,
+      id: string,
+      name: string,
+      description: string,
+      access: int,
+      ensOnly: bool,
+      color: string,
+      imageUrl: string,
+      aX: int, aY: int, bX: int, bY: int) =
+    try:
+      var image = singletonInstance.utils.formatImagePath(imageUrl)
+      let response = status_go.editCommunity(
+        id,
+        name,
+        description,
+        access,
+        ensOnly,
+        color,
+        image,
+        aX, aY, bX, bY)
+      
+      if response.error != nil:
+        let error = Json.decode($response.error, RpcError)
+        raise newException(RpcException, "Error editing community: " & error.message)
+
+      if response.result != nil and response.result.kind != JNull:
+        let community = response.result["communities"][0].toCommunityDto()
+        self.events.emit(SIGNAL_COMMUNITY_EDITED, CommunityArgs(community: community))
+    except Exception as e:
+      error "Error editing community", msg = e.msg
+      raise newException(RpcException, "Error editing community: " & e.msg)
```

Comment:
@jrainville in this and many other places here, you don't need to raise exception from the "catch" block, catch block should be used to handle potential exception, so it's enough to print error and if some returns are expected from the function to return some default value in case of exception. Raising an exception in the line 291 here is ok, since it will be caught by this catch block.

Please if you don't mind update catch block for all other methods for this service (remove raising exception).